auth/rest.go

```diff
@@ -290,3 +227,103 @@ func setCORSHeaders(headers http.Header) {
 	headers.Set("Access-Control-Allow-Origin", "*")
 	headers.Set("Access-Control-Allow-Credentials", "true")
 }
+
+func rewriteAccessTokens(resp *httptest.ResponseRecorder, req *http.Request, a *RestAuthDecorator) error {
+	err := rewriteBodyAccessTokens(resp, req, a)
+	if err != nil {
+		return err
+	}
+
+	err = rewriteHeaderAccessTokens(resp, req, a)
+	if err != nil {
+		return err
+	}
+
+	return rewriteCookieAccessTokens(resp, req, a)
+}
+
+func rewriteBodyAccessTokens(resp *httptest.ResponseRecorder, req *http.Request, a *RestAuthDecorator) error {
+	// rewrite body tokens
+	bodyTokenKey := resp.Header().Get("X-Gateway-BodyToken")
+	if bodyTokenKey != "" {
+
+		var response map[string]string
+		jsonBlob, err := ioutil.ReadAll(resp.Body)
+		if err != nil {
+			return err
+		}
+
+		err = json.Unmarshal(jsonBlob, &response)
+
+		if err != nil {
+			return err
+		}
+
+		bodyToken, ok := response[bodyTokenKey]
+
+		if !ok {
+			return err
+		}
+
+		jwtResponse := JWTResponse{}
+		jwtResponse.JWT = string(bodyToken)
+
+		token, _, err := a.tokenStore.AddToken(&jwtResponse)
+		if err != nil {
+			return err
+		}
+
+		response[bodyTokenKey] = token
+		jsonResponse, err := json.Marshal(&response)
+		if err != nil {
+			return err
+		}
+		resp.Write(jsonResponse)
+	}
+}
+
+func rewriteHeaderAccessTokens(resp *httptest.ResponseRecorder, req *http.Request, a *RestAuthDecorator) error {
+
+	headerTokenKey := resp.Header().Get("X-Gateway-HeaderToken")
+	if headerTokenKey != "" {
+
+		header := resp.Header().Get(headerTokenKey)
+
+		jwtResponse := JWTResponse{}
+		jwtResponse.JWT = string(header)
+
+		token, _, err := a.tokenStore.AddToken(&jwtResponse)
+		if err != nil {
+			return err
+		}
+
+		resp.Header().Set(headerTokenKey, token)
+	}
+
+	return nil
+}
+
+func rewriteCookieAccessTokens(resp *httptest.ResponseRecorder, req *http.Request, a *RestAuthDecorator) error {
+	cookieTokenKey := resp.Header().Get("X-Gateway-CookieToken")
+	if cookieTokenKey != "" {
+
+		cookie, err := req.Cookie(cookieTokenKey)
```

Comment:
Searching didn't yield the result I was looking for... any pointers on how to correctly get the cookie from the response ?